### DISCUSSION TITLE
M600 resume with correct fan speed if <100% + extruder display

### DISCUSSION THREAD

**Discussion by ManuMod** [Guilouz/Klipper-Flsun-Speeder-Pad#105] [Ideas]
Bonjour à tous,

Depuis peu, j'imprime majoritairement avec du PET recyclé issue de bouteilles type Evian, Quézac, Badoit, etc. Du coup, je parle désormais en mètre de filament et utilise énormément la pause sur capteur fin filament, **mais surtout une ventilation à 20, voir 25%.**

Je vais également chercher de mon côté, mais j'avoue que ce serait plus polyvalent de stocker l'état de la ventilation et de le restaurer dans la macro "Resume".

De plus, je cherche également à modifier sur l'affichage principal, le débit extrudeur (non utile pour moi) en "_filament consommé / filament total_".

A suivre du coup ;-)

**Comment by ManuMod** [comment]
Hello everybody,

I answer myself in english for sharing the solution to keep the fan_speed after a filament change pause.

- Add this line in the M600 macro for stock the value of the fan_speed (like extruder.target): 
    `SET_GCODE_VARIABLE MACRO=M600 VARIABLE=fan_speed VALUE={printer.fan.speed}`

    ![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/16747253/2da4ba85-ee33-4b87-beff-942ac18304a8)

- Add this line in the RESUME macro for reload it when resume print **and uncheck (or delete) the gcode for fan_speed to 100%**:
    `M106 S{printer["gcode_macro M600"].fan_speed}`
    `#M106 S255`

    ![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/16747253/0741dd26-2a24-433e-98d7-06972ab5b0c1)

**Comment by Guilouz** [comment]
Thanks, I have updated config files.

**Comment by ManuMod** [comment]
Je pensais la garder ouverte au cas où quelqu'un aurait une idée pour l'affichage klipperscreen du filament.
Mais c'est pas impossible que dans une motivation débordante ou en avoir encore plus besoin ou les 2 (lol), je trouve comme un grand et partage la solution.
